config/config.go

```diff
@@ -973,6 +978,8 @@ type InhibitRule struct {
 	SourceMatchRE MatchRegexps `yaml:"source_match_re,omitempty" json:"source_match_re,omitempty"`
 	// SourceMatchers defines a set of label matchers that have to be fulfilled for source alerts.
 	SourceMatchers Matchers `yaml:"source_matchers,omitempty" json:"source_matchers,omitempty"`
+	// Sources defines a set of source matchers and equal labels.
```

Comment:
The docs are incomplete and missing a lot of context, for example:

- it allows an inhibition rule to match multiple source alert
- the fact that it uses an AND operator to match multiple sources
- or that this will override `SourceMatchers`
- etc.

Also we should think about if this should become the new default and we deprecate `SourceMatchers` so a rule can have one or more of this new source matcher, if more than one all must match. You get the idea.
I'm interested to know what others think since we have already another deprecated config here, which would add more deprecations.
Maybe it's time for `InhibitRule` version 2 (or versioned config in general)?